cfme/tests/infrastructure/test_webmks_console.py

```diff
@@ -98,55 +107,31 @@ def test_webmks_vm_console(request, appliance, provider, vm_obj,
         assert (result_before_login.output.split('=')[-1].strip() <
              result_after_login.output.split('=')[-1].strip()), "Login Failed"
 
-    # This regex can find if there is a word 'login','password','incorrect' present in
-    # text, irrespective of its case
-    regex_for_login_password = re.compile(r'\blogin\b | \bpassword\b| \bincorrect\b',
-     flags=re.I | re.X)
-
-    def _validate_login():
-        """
-        Try to read what is on present on the last line in console.
-
-        If it is word 'login', enter username, if 'password' enter password, in order
-        to make the login successful
-        """
-        if vm_console.find_text_on_screen(text_to_find='login', current_line=True):
-            vm_console.send_keys(console_vm_username)
-
-        if vm_console.find_text_on_screen(text_to_find='Password', current_line=True):
-            vm_console.send_keys("{}\n".format(console_vm_password))
-        # if the login attempt failed for some reason (happens with RHOS-cirros),
-        # last line of the console will contain one of the following words:
-        # [login, password, incorrect]
-        # if so, regex_for_login_password will find it and result will not be []
-        # .split('\n')[-1] splits the console text on '\n' & picks last item of resulting list
-        result = regex_for_login_password.findall(vm_console.get_screen_text().split('\n')[-1])
-        return result == []
+        logger.info("Wait to get the '$' prompt")
 
-    # if _validate_login() returns True, it means we did not find any of words
-    # [login, password, incorrect] on last line of console text, which implies login success
-    wait_for(func=_validate_login, timeout=300, delay=5)
+        vm_console.wait_for_text(text_to_find=provider.data.templates.get('console_template')
+            ['prompt_text'], timeout=200)
 
-    logger.info("Wait to get the '$' prompt")
-
-    vm_console.wait_for_text(text_to_find=provider.data.templates.get('console_template')
-        ['prompt_text'], timeout=200)
-
-    with ssh.SSHClient(hostname=vm_obj.ip_address, username=console_vm_username,
-            password=console_vm_password) as vm_ssh_client:
         # create file on system
         vm_console.send_keys("touch blather\n")
-        wait_for(func=lambda: vm_ssh_client.run_command("ls blather", ensure_user=True) == 0,
+        wait_for(func=lambda: ssh_client.run_command("ls blather", ensure_user=True) == 0,
             delay=1, num_sec=10)
         # if file was created in previous steps it will be removed here
         # we will get instance of SSHResult
         # Sometimes Openstack drops characters from word 'blather' hence try to remove
         # file using partial file name. Known issue, being worked on.
-        command_result = vm_ssh_client.run_command("rm blather", ensure_user=True)
+        command_result = ssh_client.run_command("rm blather", ensure_user=True)
         assert command_result
+
+    except Exception as e:
```

Comment:
I'm willing to accept this in the test for now, but this should really be in the framework itself, so that any test using a console will generate console screenshots on exceptions.  Either in the exception handler plugin, or in vm_console module itself.

Reply:
I agree to this.

Reply:
Thanks @kedark3, saw you did create a card for moving this to make it more widely available.